samples/csharp_dotnetcore/80.skills-simple-bot-to-bot/SimpleRootBot/AdapterWithErrorHandler.cs

```diff
@@ -17,57 +18,96 @@ namespace Microsoft.BotBuilderSamples.SimpleRootBot
+            await sendErrorMessageAsync(turnContext, exception);
+            await endSkillConversationAsync(turnContext);
+            await clearConversationStateAsync(turnContext);
+        }
 
-                            var endOfConversation = Activity.CreateEndOfConversationActivity();
-                            endOfConversation.Code = EndOfConversationCodes.Unknown;
-                            endOfConversation.ApplyConversationReference(turnContext.Activity.GetConversationReference(), true);
+        protected async Task sendErrorMessageAsync(ITurnContext turnContext, Exception exception)
+        {
+            // Send a message to the user
+            var errorMessageText = "The bot encountered an error or bug.";
+            var errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.IgnoringInput);
+            await turnContext.SendActivityAsync(errorMessage);
 
-                            await skillClient.PostActivityAsync(botId, activeSkill, skillsConfig.SkillHostEndpoint, (Activity)endOfConversation, CancellationToken.None);
-                        }
+            errorMessageText = "To continue to run this bot, please fix the bot source code.";
+            errorMessage = MessageFactory.Text(errorMessageText, errorMessageText, InputHints.ExpectingInput);
+            await turnContext.SendActivityAsync(errorMessage);
 
-                        // Delete the conversationState for the current conversation to prevent the
-                        // bot from getting stuck in a error-loop caused by being in a bad state.
-                        // ConversationState should be thought of as similar to "cookie-state" in a Web pages.
-                        await conversationState.DeleteAsync(turnContext);
-                    }
-                    catch (Exception ex)
-                    {
-                        logger.LogError(ex, $"Exception caught on attempting to Delete ConversationState : {ex}");
-                    }
-                }
+            // Send a trace activity, which will be displayed in the Bot Framework Emulator
+            await turnContext.TraceActivityAsync("OnTurnError Trace", exception.ToString(), "https://www.botframework.com/schemas/error", "TurnError");
+        }
+
+        private async Task<InvokeResponse> endSkillConversationAsync(ITurnContext turnContext)
+        {
+            if (_conversationState == null || _skillClient == null || _skillsConfig == null)
+            {
+                return null;
+            }
 
-                // Send a trace activity, which will be displayed in the Bot Framework Emulator
-                await turnContext.TraceActivityAsync("OnTurnError Trace", exception.ToString(), "https://www.botframework.com/schemas/error", "TurnError");
-            };
+            try
+            {
+                await _conversationState.SaveChangesAsync(turnContext, true);
```

Comment:
Move the SaveChangesAsync right before the _skillClient.PostActivityAsync call, otherwise, the property you are creating below won't be persisted to state.